### ISSUE TITLE
Case that min-max normalization misfunction

### ISSUE THREAD

**Issue by Yonggie** [IvanFei/Co-Occurrence-Layer.pytorch#4]
In function``_quantization_input_as_bins`` from ``src/co_layer.py``:
When there's negative value in ``input``, there are chances that min-max normalization misfunction.
Say if tensor ``A.max()=2``, ``A.min()=-1``, then max-min ing element ``2`` is ``(2-(-1))/2``, which result to a value greater than 1, then this 1 soon is multiplied by ``num_quantization`` and then ``floor`` to index ``num_quantizetion``, which is out of range of shape.

The author of the paper wants the filter to work on real-world image, in which there're no neg values. But what if we use it in neg-possible input?
In the paper, author said:
```python
Because the input activation values can take any real
value, we quantize them uniformly into k bins. Specifically, we normalize the values of every channel to be in the range [0,1] and then for each x ∈ [0,1], we define the index of x as [x] = round(kx)
```
Does this problem also exist in the original implementation in tensorflow given by author?

**Comment by IvanFei** [comment]
hello, @Yonggie 

You're right, The correct norm function should be `(input - input_min) / (input_max - input_min)`. 

And I check the code of original implementation, the author use the correct format of norm function.  You can ref https://github.com/Shevlev/Co-occurrence-Neural-Network/blob/master/Toy%20Example/co_layer.py#L133-L142

best wishes,
IvanFei
